src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -42,11 +44,17 @@ public void shouldSupportConfigurationAsCode() throws Exception {
         String config = resource.toString();
         ConfigurationAsCode.get().configure(config);
 
+        ExternalWorkspaceProperty.DescriptorImpl exwsNodeProperty = ExtensionList.lookupSingleton(ExternalWorkspaceProperty.DescriptorImpl.class);
+        List<NodeDiskPool> nodeDiskPools = exwsNodeProperty.getNodeDiskPools();
```

Comment:
The above code does not work because the descriptor does not have a list of `NodeDiskPools`:

https://github.com/jenkinsci/external-workspace-manager-plugin/blob/08073fc5c2878633f5129b6d583b83c5263274a7/src/main/java/org/jenkinsci/plugins/ewm/nodes/ExternalWorkspaceProperty.java#L35-L44

I don't really understand how to fix it... descriptors are still a mystery to me.

Reply:
I think in this case we need to do some refactoring by moving the `List<NodeDiskPool> nodeDiskPools` from `ExternalWorkspaceProperty` class to the `DescriptorImpl`. I'm not very sure what will break by doing this 🤔 

I'll try to look these days over the required changes to be able to do such refactoring.

Reply:
So you can change the `ExternalWorkspaceProperty` like following, but I'm not sure if this is backwards compatible with previous plugin versions ❗️ 

```java
public class ExternalWorkspaceProperty extends NodeProperty<Node> {

    @DataBoundConstructor
    public ExternalWorkspaceProperty(List<NodeDiskPool> nodeDiskPools) {
        this.getDescriptor().setDiskPools(fixNull(nodeDiskPools));
    }

    @Override
    public DescriptorImpl getDescriptor() {
        return (DescriptorImpl) super.getDescriptor();
    }

    @Nonnull
    public List<NodeDiskPool> getNodeDiskPools() {
        return getDescriptor().getNodeDiskPools();
    }

    @Extension
    @Symbol("exwsNodeConfigurationDiskPools")
    public static class DescriptorImpl extends NodePropertyDescriptor {

        private List<NodeDiskPool> nodeDiskPools = Collections.emptyList();

        @Nonnull
        @Override
        public String getDisplayName() {
            return Messages.nodes_ExternalWorkspaceProperty_DisplayName();
        }

        @Nonnull
        public List<NodeDiskPool> getNodeDiskPools() {
            return Collections.unmodifiableList(nodeDiskPools);
        }

        @DataBoundSetter
        public void setDiskPools(List<NodeDiskPool> nodeDiskPools) {
            this.nodeDiskPools = nodeDiskPools;
        }
    }
}
```

Reply:
It works for the test, but breaks backward compatibility. The ConfigMigrationTest fails.

Reply:
I am thinking the NodeDiskPool class also needs to change. Descriptors are so mysterious to me.

Reply:
Looks like NodeDisk also has to change.

Reply:
Do we really need to change `NodeDiskPool` and `NodeDisk`? I think it should work even without changing those.

Reply:
Well, I was trying to get the test to pass, and it was failing instead, that's how these recent changes started. Maybe the test needs to be written differently.

Reply:
For me, I think we might not read ExternalWorkspaceProperty like this. There should be a class containing all nodeProperties, and we suppose that class is NodeProperties. Then we do something like ```NodeProperties.getDescriptor().getProperties(ExternalWorkspaceProperty.class) ``` .(The syntax is wrong, but should be very same, I'm working on finding that class)